### PR TITLE
FF131 Secure attr must be set if SameSite=None

### PR DESCRIPTION
**Pull request by hamishwillee**
FF131 enables preference by default such that if `SameSite=None` the  `Secure` attribute must be set in https://bugzilla.mozilla.org/show_bug.cgi?id=1909673

Unfortunately this required a little more than just replacing the pref subfeature with a version.

The pref was named "secure_context_required", which is incorrect for HTTP headers (it's reasonable for APIs though). The requirement is actually about what headers must be set, so I have renamed this to `secure_required_if_none` and put a proper description. Could also have moved this below the `None` option, but I think this level is better.

It would be good if we also had an entry for the `Secure` attribute, at the same level as `HttpOnly`. However I do not have support data for that.

Related docs work can be tracked in https://github.com/mdn/content/issues/35697

### CODE REVIEW

**Comment by hamishwillee** [comment]
@Elchi3 @queengooborg Could I please get a review on this - I did it about a month ago and the release is today.

File: http/headers/Set-Cookie.json [hunk 1]
```diff
@@ -449,32 +443,31 @@
               }
             }
           },
-          "schemeful": {
+          "secure_required_if_none": {
```
Comment (caugner) [inline http/headers/Set-Cookie.json:446]: Nit: Have you considered calling this `none_requires_secure`?

**Review by caugner — APPROVED** [review]
LGTM, but left one comment to consider. Feel free to ignore and merge.

File: http/headers/Set-Cookie.json (hunk 1 above)
Comment (hamishwillee) [reply http/headers/Set-Cookie.json:446]: No, but I like your key better.
```suggestion
          "none_requires_secure": {
```

**Pull request merged**